doc/use_cases.rst

```diff
@@ -109,3 +109,59 @@ line.
 .. include:: examples/install_extension.py
    :code: python
    :start-line: 16
+
+=========================================================================
+Get a list of available packages filtered by their relation to the system
+=========================================================================
+
+A system user wants to obtain a list of available RPM packages for their
+consecutive automatic processing or for informative purpose only.
+The list of RPM packages is filtered by requested relation to the system
+or user provided <package-name-specs>. The obtained list of packages
+is based on available data supplied by accessible system repositories.
+
+A relation to the system might be for example one of the following:
+
+installed - packages already installed on the system
+
+available - packages available in any accessible repository
+
+extras - packages installed on the system not available in any known
+repository
+
+obsoletes - installed packages that are obsoleted by packages in any
+accessible repository
+
+recent - packages recently added into accessible repositories
+
+upgrades - available packages upgrading some installed packages
+
+-----
+CLI
+-----
+
+::
+
+    dnf list [all] [<package-name-specs>...]
```

Comment:
I'd prefer if these examples could be just copy&pasted into a terminal by a user. So, I'd remove ```[all]``` and I'd replace ```[<package-name-specs>...]``` by some real-world examples. The usage is already available in man pages - no need to duplicate it.